File: backend/src/main/java/com/overcode/model/Player.java
package com.overcode.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class Player {

    private Long id;
    private Long externalId;

    private String name;
    private Integer currentPrice;
    private String clubName;

    private Integer goals;
    private Integer assists; // no se muestra
    private Integer shotsOnTarget;
    private Integer passes;

    private Integer interceptions;
    private Integer tackles;
    private Integer keyPasses; // no se muestra
    private Double rating;

    private Integer successfulDribbles;

    private List<Token> tokens;

    public Player(Long id, String name) {
        this.id = id;
        this.name = name;
        setCurrentPrice(1);
        setTokens(getInitialTokens()); // deberían ser 100 tokens acá
    }

    private List<Token> getInitialTokens() {
        List<Token> newTokens = new java.util.ArrayList<>(List.of());
        for (int i = 0; i < 100; i++) {
            newTokens.add(new Token(this));
        }
        return newTokens;
    }

    public Player(String name) {
        this(null, name);
    }

    public Player(Long id, String name, Integer currentPrice, List<Token> tokens) {
        this.id = id;
        this.name = name;
        setCurrentPrice(currentPrice);
        setTokens(tokens);
    }

    public void setCurrentPrice(Integer currentPrice) {
        this.currentPrice = Math.max(currentPrice, 1);
    }
}

File: frontend/src/types/player.ts
export interface Player {
    id: number;
    name: string;
    currentPrice: number;
    clubName: string;
    goals: number;
    shotsOnTarget: number;
    passes: number;
    tackles: number;
    rating: number;
    interceptions: number;

    // tokens:number[];
}
// Long id,
//     String name,
//     Integer currentPrice,
//     String clubName,
//     Integer goals,
//     Integer shotsOnTarget,
//     Integer passes,
//     Integer interceptions,
//     Integer tackles,
//     Double rating,
// List<TokenResponseDTO> tokens